test/pom.xml

```
@@ -270,6 +270,15 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <configuration>
+          <excludes>
+            <exclude>**/hadoop-metrics2-accumulo.properties</exclude>
```

It occurs to me... does this still allow metrics to work when we're doing testing? If it's not in the jar, how does it get on the class path?

---

I was under the impression that the maven-jar-plugin would be called during assembly (package?)  If you build / test under maven, the property file will still be there.  So, any testing that uses classes / files under ..../target/...  should still work.

You are correct though, if you tried to run it tests using just the jars - unless you provided a metrics configuration file with metrics output configured, then the tests that check those files would fail. If you go to the trouble to run the tests outside of maven using a command line, then if you got that far, adding a suitable config file may not be much of a hurdle and forces you to provide a configuration that you want for your needs.

One issue is that the metrics config file is specifying ./target/name as a know location for the output files - this was so that mvn clean would remove them and they would not be left hanging around. 

I suppose that if there was another well know location that would always be suitable for writes, then that could be used instead.  Sounds like just specified /tmp, but that's making an assumption that /tmp would always be suitable - for example would docker / containers / aws / ... prefer not to use /tmp but something else?  And then there's the clean-up - would need something besides (or modification) mvn clean.  Sure, /tmp may eventually get cleaned, but seems good practice not to depend on that.

---

`maven-jar-plugin` runs during the `package` phase of the default lifecycle. Usually, the `maven-assembly-plugin` is also configured to run during the `package` phase, but only if a user manually configured it, since it's not bound to the default lifecycle by default.

If our tests pass, I have no problem excluding it from the jar. Like you said, our metrics file is set for target directory, specifically, so it's clear it's only for our tests anyway. And, if our test code were reused, it wouldn't be hard for the caller to provide its own metrics config file.

I would avoid `/tmp`. It's best to use the `target/` directory, and write to no other directories in a Maven build.

My final point of feedback, then, is just a suggestion to use the path explicitly, rather than wildcards:

```suggestion
            <exclude>src/main/resources/hadoop-metrics2-accumulo.properties</exclude>
```

---

Hit commit via ui, becuase - sure that's reasonable - and then decided to test - oops.  Anyway I don't think that works as intended - might be src is assumed?  Anyway, checking now...

---

Oh! The jar plugin's base directory might be target/classes, where the compiler plugin leaves class files and the resource plugin leaves resources. It might just work with the file name only, since this resource isn't in a subdirectory. Not sure.

---

So I basically tried more restrictive paths combinations and **/filename seems to be the "most" specific that actually excluded the file from the created jar - basically revered the UI generated commit. I examined the jars created with the current changes and run -Psunny and things look okay as it stands now.

---

I tested the following more narrow construction, and it seems to work fine (see my last comment for why):

```suggestion
            <exclude>hadoop-metrics2-accumulo.properties</exclude>
```